app/uk/gov/hmrc/emcstfereferencedata/connector/retrieveCnCodeInformation/RetrieveCnCodeInformationConnectorCRDL.scala

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2025 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.emcstfereferencedata.connector.retrieveCnCodeInformation
+
+/*
+ * Copyright 2025 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import play.api.Logger
+import uk.gov.hmrc.emcstfereferencedata.models.request.CnInformationRequest
+import uk.gov.hmrc.emcstfereferencedata.models.response.{CnCodeInformation, ErrorResponse}
+import uk.gov.hmrc.emcstfereferencedata.repositories.CnCodesRepository
+import uk.gov.hmrc.http.HeaderCarrier
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class RetrieveCnCodeInformationConnectorCRDL @Inject() (
+  repository: CnCodesRepository
+) extends RetrieveCnCodeInformationConnector {
+
+  lazy val logger: Logger = Logger(this.getClass)
+
+  override def retrieveCnCodeInformation(cnInformationRequest: CnInformationRequest)(implicit
+    hc: HeaderCarrier,
+    ec: ExecutionContext
+  ): Future[Either[ErrorResponse, Map[String, CnCodeInformation]]] =
+    repository
+      .fetchCnCodeInformation(cnInformationRequest)
+      .map(Right(_))
+      .recover {
+        case exception: Exception => {
+          logger.warn(
+            s"[RetrieveCnCodeInformationConnectorCRDL][retrieveCnCodeInformation] Unexpected Error fetching data from repository"
+          )
```

Comment:
```suggestion
          logger.warn(
            s"[RetrieveCnCodeInformationConnectorCRDL][retrieveCnCodeInformation] Unexpected Error fetching data from repository",
            exception
          )
```
Oh this is nice 💯 We can also pass the `exception` to the `logger` call to provide even more context